### PR TITLE
Index logging and counter cleanup

### PR DESCRIPTION
**Pull request by lintool**
+ Tweaked logging output during indexing to make less noisy.
+ Documentation and clarification of what each counter means.
+ Better logging of final counter states.

### CODE REVIEW

File: src/main/java/io/anserini/index/IndexCollection.java [hunk 1]
```diff
@@ -152,18 +175,16 @@ public void run() {
...
             counters.unindexableDocuments.incrementAndGet();
             continue;
           }
 
           @SuppressWarnings("unchecked") // Yes, we know what we're doing here.
           Document doc = transformer.createDocument(d);
-
           if (doc != null && (whitelistDocids == null || whitelistDocids.contains(d.id()))) {
             if (args.uniqueDocid) {
               writer.updateDocument(new Term("id", d.id()), doc);
```
Comment (Peilin-Yang) [inline src/main/java/io/anserini/index/IndexCollection.java]: What about the `else` case? Should we increment `counters.unindexedDocuments` here?

File: src/main/java/io/anserini/index/IndexCollection.java [hunk 2]
```diff
@@ -152,31 +182,36 @@ public void run() {
         while (iter.hasNext()) {
           SourceDocument d = iter.next();
           if (d == null) {
-            LOG.info("Null document encountered");
+            counters.unindexed.incrementAndGet();
```
Comment (Peilin-Yang) [inline src/main/java/io/anserini/index/IndexCollection.java]: So I think this is a problem: 
Currently, if we encounter null document we do not know if it is EOF or the real null document (b/c of errors or sth). This is mainly due to how `Collection` read the documents. 
I think we should remove this for now and revisit it in #254

File: src/main/java/io/anserini/index/IndexCollection.java [hunk 3]
```diff
@@ -300,13 +335,21 @@ public void run() throws IOException, InterruptedException {
       }
     }
 
-    LOG.info("Indexed documents: " + counters.indexedDocuments.get());
-    LOG.info("Empty documents: " + counters.emptyDocuments.get());
-    LOG.info("Errors: " + counters.errors.get());
+    if (numIndexed != counters.indexed.get()) {
```
Comment (Peilin-Yang) [inline src/main/java/io/anserini/index/IndexCollection.java]: Is this tested for Microblog13 corpus with index option `uniqueDocid`?

**Comment by lintool** [comment]
@Peilin-Yang Addressed CR. Running from scratch:
```
nohup python src/main/python/run_regression_tweets2013.py >& log.tweets2013 &
```

**Comment by lintool** [comment]
@Peilin-Yang Actually, I just realized `run_regression_tweets2013.py` is likely to fail... let me rethink how to handle.

**Comment by lintool** [comment]
@Peilin-Yang Actually, you were correct. For Tweets2013 the indexer did throw an exception because of the duplicate docs issue. However, since by that time the index had already been built, the exception doesn't really do anything and the regression completes as usual.

I've changed to logging a warning instead of throwing an exception. Running one final time and then I'll merge.

**Comment by Peilin-Yang** [comment]
I am a bit confused. Is this related to #282 ?

**Comment by lintool** [comment]
@Peilin-Yang This is related to your `-uniqueDocid` comment.
Regardless, tweets2011 and tweets2013 passed perfectly.

**Pull request merged**